packit_service/worker/checker/copr.py

```diff
@@ -138,3 +141,44 @@ def _pre_check(self) -> bool:
                 )
                 return False
         return True
+
+
+class AreFilesChanged(Checker, GetCoprBuildJobHelperForIdMixin):
+    """
+    Check if any files under the current package's `paths` field is changed.
+    If not, then just skip the current copr build job.
+    """
+
+    def get_files_changed(self) -> list[Path]:
+        """
+        Get the list of files changed in the current commit or the current pullrequest
+        """
+        # Get the changes object
+        if self.job_config.trigger == JobConfigTriggerType.pull_request:
+            changes = self.copr_build_helper.pull_request_object.changes
+        elif self.job_config.trigger == JobConfigTriggerType.commit:
+            branch = self.job_config.branch or self.project.default_branch
+            changes = self.copr_build_helper.project.get_commit(branch)
+        else:
+            raise AssertionError(f"Trigger not supported: {self.job_config.trigger}")
+        # The paths here are relative to the git repo root
+        return [Path(file) for file in changes.files]
+
+    def pre_check(self) -> bool:
+        if self.job_config.trigger == JobConfigTriggerType.release:
+            # For releases we don't do any checks
+            return True
+        # FIXME: This is probably unnecessary
+        package_config = self.package_config.get_package_config_for(self.job_config)
+        # The paths that we need to check for files changed
+        paths = package_config["paths"]
+        # Early check if the git root was included, in which case we don't need to
+        # check the files changed
+        # TODO: refine this check when gitignore-like patterns are supported
+        if "./" in paths:
+            return True
+        for changed_file in self.get_files_changed():
+            # Check if any of the files changed are under the paths that are being tracked
+            if any(changed_file.is_relative_to(p) for p in paths):
+                return True
+        return False
```

Comment:
Actually, what do we do if we have an empty commit, like the Fedora rebuild commits? Should it be a special case to skip this check?